library/core/src/ptr/mut_ptr.rs

```diff
@@ -2302,3 +2314,90 @@ impl<T: ?Sized> PartialOrd for *mut T {
         *self >= *other
     }
 }
+
+#[cfg(kani)]
+#[unstable(feature = "kani", issue = "none")]
+mod verify {
+    use crate::kani;
+    use kani::PointerGenerator;
+    use core::mem;
+
+    // The proof for a unit type panics as offset_from needs the pointee size > 0
+    #[kani::proof_for_contract(<*mut ()>::offset_from)]
+    #[kani::should_panic]
+    pub fn check_mut_offset_from_unit() {
+        let mut val: () = ();
+        let src_ptr: *mut () = &mut val;
+        let dest_ptr: *mut () = &mut val;
+        unsafe {
+            dest_ptr.offset_from(src_ptr);
+        }
+    }
+
+    // Array size is bound for kani::any_array
+    const ARRAY_LEN: usize = 40;
```

Comment:
Why 40? Let's add a justification for any magic number (e.g., this keeps the verification time under X and it's enough to cover all code in the target functions).

Reply:
Added comments explaining the size in a latest commit.